### PR TITLE
Tidy up path handling in document_file

### PR DESCRIPTION
**Pull request by TrueDoctor**
* Improve solution for #455
* Reduce the number of allocations in document_file.rs

### CODE REVIEW

File: editor/src/communication/dispatcher.rs [hunk 1]
```diff
@@ -339,10 +339,19 @@ mod test {
 		init_logger();
 		let mut editor = create_editor_with_three_layers();
 
-		let sorted_layers = editor.dispatcher.documents_message_handler.active_document().all_layers_sorted();
+		fn to_vec<'a>(paths: Vec<&'a [LayerId]>) -> Vec<Vec<LayerId>> {
```
Comment (0HyperCube) [inline editor/src/communication/dispatcher.rs:342]: A bit confusing with the std `.to_vec` (copying outer vec). Perhaps `map_to_vec`?

File: editor/src/communication/dispatcher.rs [hunk 2]
```diff
@@ -339,10 +339,19 @@ mod test {
...
+			paths.iter().map(|layer| layer.to_vec()).collect::<Vec<_>>()
+		}
+		let sorted_layers = to_vec(editor.dispatcher.documents_message_handler.active_document().all_layers_sorted());
 		println!("Sorted layers: {:?}", sorted_layers);
```
Comment (0HyperCube) [inline editor/src/communication/dispatcher.rs:346]: Probably should be debug / trace.

Comment (TrueDoctor) [reply editor/src/communication/dispatcher.rs:346]: Thought the same thing, but afaik this is only used for debugging test cases which use std out

Comment (TrueDoctor) [reply editor/src/communication/dispatcher.rs:346]: We could put it into a #[cfg(test)] though

File: editor/src/communication/dispatcher.rs (hunk 1 above)
Comment (TrueDoctor) [reply editor/src/communication/dispatcher.rs:342]: yeah wasn't happy about the name either

**Review by otdavies — APPROVED** [review]
Heading read over this PR with @TrueDoctor, tested it locally and gone over it multiple times, it seems like an overall positive improvement. It has my go-ahead!

**Pull request merged**